pydantic/_internal/_typing_extra.py

```diff
@@ -225,17 +225,54 @@ def parent_frame_namespace(*, parent_depth: int = 2, force: bool = False) -> dic
     return frame.f_locals
 
 
+def _type_convert(arg: Any) -> Any:
+    """Convert `None` to `NoneType` and strings to `ForwardRef` instances.
+
+    This is a backport of the private `typing._type_convert` function. When
+    evaluating a type, `ForwardRef._evaluate` ends up being called, and is
+    responsible for making this conversion. However, we still have to apply
+    it for the first argument passed to our type evaluation functions, similarly
+    to the `typing.get_type_hints` function.
+    """
+    if arg is None:
+        return NoneType
+    if isinstance(arg, str):
+        # Like `typing.get_type_hints`, assume the arg can be in any context,
+        # hence the proper `is_argument` and `is_class` args:
+        return _make_forward_ref(arg, is_argument=False, is_class=True)
+    return arg
+
+
+@overload
 def get_cls_type_hints(
-    obj: type[Any], *, ns_resolver: NsResolver | None = None, lenient: bool = False
-) -> dict[str, Any]:
+    obj: type[Any],
+    *,
+    ns_resolver: NsResolver | None = None,
+    lenient: Literal[True],
+) -> dict[str, tuple[Any, bool]]: ...
+@overload
+def get_cls_type_hints(
+    obj: type[Any],
+    *,
+    ns_resolver: NsResolver | None = None,
+    lenient: Literal[False] = ...,
+) -> dict[str, Any]: ...
+def get_cls_type_hints(
+    obj: type[Any],
+    *,
+    ns_resolver: NsResolver | None = None,
+    lenient: bool = False,
```

Comment:
Why not `Literal[False]` to be consistent with the above?

Reply:
It's specified in the second overload